app/src/main/java/com/fmsys/snapdrop/MainActivity.java

```diff
@@ -189,15 +189,17 @@ protected void onCreate(final Bundle savedInstanceState) {
         CookieManager.getInstance().setAcceptThirdPartyCookies(binding.webview, true);
 
         binding.webview.setDownloadListener((url, userAgent, contentDisposition, mimetype, contentLength) -> {
+            Log.d("download listener", "search file of size " + contentLength);
             synchronized (downloadFilesList) {
                 final Iterator<JavaScriptInterface.FileHeader> iterator = downloadFilesList.iterator();
                 while (iterator.hasNext()) {
                     final JavaScriptInterface.FileHeader file = iterator.next();
-                    if (file.getSize().equals(String.valueOf(contentLength))) {
```

Comment:
For some reason `contentLength` was -1 in my emulator, so I removed the check for now...